homeassistant/components/openweathermap/sensor.py

```diff
@@ -1,123 +1,122 @@
+    """Set up OpenWeatherMap sensor entities based on a config entry."""
+    domain_data = hass.data[DOMAIN][config_entry.entry_id]
+    name = domain_data[ENTRY_NAME]
+    weather_coordinator = domain_data[ENTRY_WEATHER_COORDINATOR]
+    forecast_coordinator = domain_data[ENTRY_FORECAST_COORDINATOR]
+    monitored_conditions = domain_data[ENTRY_MONITORED_CONDITIONS]
+
+    sensor_types = _get_sensor_types(hass.config.units)
+
+    entities = []
+    for sensor_type in monitored_conditions:
+        unique_id = f"{config_entry.unique_id}-{sensor_type.lower()}"
+        entities.append(
+            OpenWeatherMapSensor(
+                name,
+                unique_id,
+                sensor_type,
+                sensor_types,
+                weather_coordinator,
+                forecast_coordinator,
+            )
         )
 
-    add_entities(dev, True)
+    async_add_entities(entities, False)
 
 
 class OpenWeatherMapSensor(Entity):
     """Implementation of an OpenWeatherMap sensor."""
 
-    def __init__(self, name, weather_data, sensor_type, temp_unit):
+    def __init__(
+        self,
+        name,
+        unique_id,
+        sensor_type,
+        sensor_types,
+        weather_coordinator: WeatherUpdateCoordinator,
+        forecast_coordinator: ForecastUpdateCoordinator,
+    ):
         """Initialize the sensor."""
-        self.client_name = name
-        self._name = SENSOR_TYPES[sensor_type][0]
-        self.owa_client = weather_data
-        self.temp_unit = temp_unit
-        self.type = sensor_type
-        self._state = None
-        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._name = name
+        self._unique_id = unique_id
+        self._sensor_type = sensor_type
+        self._sensor_name = sensor_types[sensor_type][SENSOR_NAME]
+        self._unit_of_measurement = sensor_types[sensor_type][SENSOR_UNIT]
+        self._device_class = sensor_types[sensor_type][SENSOR_DEVICE_CLASS]
+        self._weather_coordinator = weather_coordinator
+        self._forecast_coordinator = forecast_coordinator
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self.client_name} {self._name}"
+        return f"{self._name} {self._sensor_name}"
+
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return self._unique_id
+
+    @property
+    def should_poll(self):
+        """Return the polling requirement of the entity."""
+        return False
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def device_class(self):
+        """Return the device_class."""
+        return self._device_class
 
     @property
     def state(self):
         """Return the state of the device."""
-        return self._state
+        return self._weather_coordinator.data[self._sensor_type]
```

Comment:
good point,
 I suppose I can't put whole forecast into state, it can be only in attribute?

Reply:
An idea:
Create another coordinator named ex. forecast_coordinator_sensor (maybe you find a better name) and this coordinator take only the first weather object of the forecast_coordinator and provides the data in the same schema as the weather_coordinator. As the schema is equal we can use OpenWeatherMapSensor and get our defined conditions of the forecast.
We have to create an option, where the user can specify which conditions he want to monitor in the forecast. As I plan to add the One Call Feature to HA, we should think probably about a general solution, because the OneCall requests contains the hourly forecast for 48h and the daily forcast for the next seven days.

What you think about this idea?

Reply:
Ok, I got it, there will be new option with comma separated forecast condition.
For each forecast condition we create new sensor with value for this day (first object).
Yeah, can do that

Reply:
I believe `monitored_conditions` is not accepted anymore with integrations using config flow. The integration should create all possible sensors (entities) and the user can disable sensors that are not wanted. This will make it easier than having the user to find out the names of the conditions and enter them manually.

Reply:
Nice idea, can get rid of all these silly validation and conversion for conditions

Reply:
Should I need to somehow handle the disabled entity or this is responsibility of HA?

Reply:
No, this will be handled by HA.

Reply:
fixed, now by default all sensors created also created all sensors for all forecast attribute

Reply:
as this behaviour for sensors will be confirmed I will update documentation